CMakeLists.txt

```diff
@@ -267,16 +267,23 @@ endif()
 # Make sure the generated paths exist
 execute_process(COMMAND mkdir -p "${CMAKE_BINARY_DIR}/generated")
 
-# We need to link some packages as dynamic/dependent.
-set(CMAKE_FIND_LIBRARY_SUFFIXES .dylib .so)
-find_package(OpenSSL REQUIRED)
-find_package(BZip2 REQUIRED)
-find_package(Dl REQUIRED)
-find_package(Readline REQUIRED)
+# Apple has deprecated OpenSSL, static link with brew bottled libressl
+if(APPLE)
+  set(CMAKE_FIND_LIBRARY_SUFFIXES .a)
```

Comment:
We should wrap this with:

```cmake
if(NOT DEFINED ENV{BUILD_LINK_SHARED})
  set(CMAKE_FIND_LIBRARY_SUFFIXES .a .dylib .so)
endif()
```

And... add a define for `BUILD_LINK_SHARED=1` to Homebrew's osquery formula. So building/installing through homebrew will link libre dynamically.